### ISSUE TITLE
[BUG] 'ASFDataDload' has no attribute 'bulk_downloader'

### ISSUE THREAD

**Issue by rzinke** [aria-tools/ARIA-tools#229]
**The bug**
When I try to download new data from ASF, I get an error message saying that the ASFDataDload module does not have a bulk downloader attribute.

```
(ARIA-tools) [rzinke@sar ARIA-tools]$ ariaDownload.py -t 121 -w /u/sar-r2/rzinke/Tibet/Track121/AllProducts

https://api.daac.asf.alaska.edu/services/search/param?asfplatform=Sentinel-1%20Interferogram%20(BETA)&processingLevel=GUNW_STD&output=JSON&relativeOrbit=121
Attempting to obtaining user/pass from .netrc
Traceback (most recent call last):
  File "/u/sar-r0/rzinke/python/miniconda3/envs/ARIA-tools/bin/ariaDownload.py", line 312, in <module>
    Downloader(inps)()
  File "/u/sar-r0/rzinke/python/miniconda3/envs/ARIA-tools/bin/ariaDownload.py", line 101, in __call__
    prod_dl(self.inps.num_threads)
  File "/u/sar-r0/rzinke/python/miniconda3/envs/ARIA-tools/bin/ariaDownload.py", line 269, in prod_dl
    downloader  = AD.bulk_downloader()
AttributeError: module 'ASFDataDload' has no attribute 'bulk_downloader'
```

**To Reproduce**
```
$ ariaDownload.py -t 121
```

**Desktop (please complete the following information):**
 - ARIA-tools git tag: 
commit 8411f3d5e2ad94beb830c771ed0201f3ab02dcc9 (HEAD -> dev, aria-tools/dev)
Author: BB <bbuzz318@icloud.com>
Date:   Thu Sep 17 09:10:13 2020 -0700
    fix problem for newer version (#227)
    * fix problem for newer version
    * Update tools/bin/ariaDownload.py
    codacy fix
    * Update tools/bin/ariaDownload.py
    Co-authored-by: dbekaert <bekaertdavid@gmail.com>

 - OS
JPL SAR (CentOS)

**Additional context**
I did a hard reset to ARIA-tools/dev, then removed and reinstalled the ARIA-tools conda environment, and ran the setup.py routines.

**Comment by dbekaert** [comment]
@bbuzz31 suspect this is a change on the ASF side in their bulk download python script. It would make our life easier if its a simple library. @piyushrpt expressed some similar concerns when trying to get the virtual data access to work in ISCE, which broke due to changes on the bulk download python script too.

**Comment by bbuzzanga** [comment]
@dbekaert yes they make it very difficult for us. @rzinke I think the problem has something to do with the amount of products. For now, if you split the track into two time periods the download works for me:
```
ariaDownload.py -t 121 -s 20140101 -e 20180101
ariaDownload.py -t 121 -s 20180101 -e 20210101

**Comment by rzinke** [comment]
Weird. It is working with the specified date ranges. 
Thanks!

**Comment by EJFielding** [comment]
I was working with some of the ASF folks in August to prepare the Los Angeles time series Jupyter notebook using the `ariaDownload.py` script within OpenSARLab. I was downloading the more than 700 products on one of the tracks, and I saw that the first 100 or so were downloaded at about 3 seconds each, but then the download rate started dropping and the time per product increased to 12 seconds and then 20 seconds by about product 300 and finally over 35 seconds each for the products after about product number 500. It was not clear whether this was something in the ASF server code or something in the AWS cloudfront web serving system. It is possible they now changed it to limit the number of products in one call.

At one point, I could see the download rate was so slow that the last hundred or so products were going to take another hour, so I killed my download job and then restarted it. The new download got those last products in a few minutes. It is great that the `ariaDownload.py` script (or what it calls) checks for data that already exists locally and does not repeat the download.

**Comment by bbuzzanga** [comment]
thanks for the comments @EJFielding, it would be nice if they could let us know of any such changes

**Comment by EJFielding** [comment]
The ASF people I was working with (mostly @fjmeyer and @emlundell) seemed surprised by the ASF data download slowing also, so I don't know if they could tell us about this latest change.

**Comment by emlundell** [comment]
I've been talking with the software engineers over Vertex, the search API, and some AWS resources. We're trying to figure out the root cause of the slowdowns.

However, they did also mention that for the search API call

>  output=kmz and output=url are both invalid choices that will lead to 400 BAD REQUEST or whatever the error is

and that

> Valid options are metalink, json, geojson, csv, kml, download, count
> Additionally, if they already have a list of products from a search, they can just pass the URLs directly to bulk-download.asf.alaska.edu and skip the extra search step. 

I hope this improves ARIA.

**Comment by bbuzzanga** [comment]
Hi @emlundell, thanks for looking into this.
We internally handle `kmz` and `url` outputs; they don't get passed to the API.

As far as the current issue, the bug can be traced to a 504 (Gateway timeout) from our POST to ASF.
We POST as follows:
```
import requests
url            = 'https://api.daac.asf.alaska.edu/services/search/param?&output=Download'
dict_prod =  {'product_list' : 'S1-GUNW-D-R-121-tops-20200328_20200304-001048-31205N_29218N-PP-d8c5-v2_0_2,S1-GUNW-D-R-121-tops-20200328_20200316-001048-31205N_29218N-PP-aab6-v2_0_2'}
response = requests.post(url, data=dict_prod)
if response.ok:
    script = response.text
else:
    print (response)
```

When the `product_list` gets very large, the POST fails.
I have tried increasing the timeout error on the POST but it has no effect.

Thanks for your help!

**Comment by dbekaert** [comment]
Retrieving the list of files allows us to add additional functionality not
included in the DAAC API to sort on temporal baselines etc.

**Comment by dbekaert** [comment]
So there are two different items here:
1) hammering the ASF DAAC API to retrieve a long list of products with the time-out message that @bbuzz31 provided
2) The download speed that get throttled once the download is going

**Comment by emlundell** [comment]
> The download speed that get throttled once the download is going 

I did various runs of `ariaDownload.py -b '33.5 34.5 -119.0 -117.9' --track 71` on both OpenSARlab and a local ASF server. The graphs show what happened. 

![Image Pasted at 2020-9-30 20-12](https://user-images.githubusercontent.com/12416781/94873166-f4480500-03fa-11eb-8cab-8cc9086bd63c.png)

Yesterday, I did three runs on OSL between 1.5 and 0.5 hours apart. For all three the download speeds decreased step-wise over time. I started top think that perhaps it was a throttling issue with GRFN API or the EBS volume. Looking at the GRFN API metrics, there was no unusual errors or behavior that I could see. I might have missed something but I don't think it is the GRFN API.

Looking at OSL metrics, there was no unusual CPU/RAM/network issues that I noticed. If it was the EBS volume throttling, it didn't show up in the metrics/logs. 

![Image Pasted at 2020-10-1 15-17](https://user-images.githubusercontent.com/12416781/94873183-ff9b3080-03fa-11eb-9943-cb6ad6955a80.png)

Today, I decided to run the command on a local ASF server. If there was no stepping in the download speeds then perhaps it was a EBS throttling issue. While significantly slower in speeds, the graphs lines (blue and green) are flat. Then I decided to run the OSL tests again - this time one right after another. Sometimes the EBS slowdowns are bursty and recharge over time. As the other two graph lines show, things are flat and uneventful compared to yesterday. I have no idea what caused the differences.

I'm at a lost right now. So for this problem, I'm holding off any more investigations unless the problem really starts to cause more people grief and needs to be addressed.

> hammering the ASF DAAC API to retrieve a long list of products with the time-out message that @bbuzz31 provided

Now this is still an outstanding issue. Is this fully repeatable or was it transient?
